include/rest-request.php

```diff
@@ -50,23 +55,41 @@ class PLL_REST_Request extends PLL_Base {
 	public function init() {
 		parent::init();
 
-		if ( $this->model->get_languages_list() ) {
-
-			/** This action is documented in include/class-polylang.php */
-			do_action( 'pll_no_language_defined' ); // To load overridden textdomains.
+		if ( ! $this->model->get_languages_list() ) {
+			return;
+		}
 
-			$this->filters_links = new PLL_Filters_Links( $this );
-			$this->filters = new PLL_Filters( $this );
-			$this->filters_widgets_options = new PLL_Filters_Widgets_Options( $this );
+		if ( ! empty( $_REQUEST['lang'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			if ( is_string( $_REQUEST['lang'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+				$this->curlang = $this->model->get_language( sanitize_key( $_REQUEST['lang'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
+			}
 
-			// Static front page and page for posts.
-			if ( 'page' === get_option( 'show_on_front' ) ) {
-				$this->static_pages = new PLL_Static_Pages( $this );
+			if ( empty( $this->curlang ) && ! empty( $this->options['default_lang'] ) && is_string( $this->options['default_lang'] ) ) {
+				// A lang has been requested but it is invalid, let's fall back to the default one.
+				$this->curlang = $this->model->get_language( sanitize_key( $this->options['default_lang'] ) );
 			}
+		}
 
-			$this->links = new PLL_Admin_Links( $this );
+		if ( ! empty( $this->curlang ) ) {
+			$GLOBALS['text_direction'] = $this->curlang->is_rtl ? 'rtl' : 'ltr';
```

Comment:
As we don't flter the locale, we should not change the text direction.

Reply:
Done in a6700ee.